pyro/nn/auto_reg_nn.py

```diff
@@ -52,22 +52,23 @@ def create_mask(input_dim, observed_dim, hidden_dims, permutation, output_dim_mu
 
     # Create the indices that are assigned to the neurons
     input_indices = torch.cat((torch.zeros(observed_dim), 1 + var_index))
-    hidden_indices = [sample_mask_indices(input_dim, h) for h in hidden_dims]
+    if observed_dim > 0:
+        hidden_indices = [sample_mask_indices(input_dim, h)-1 for h in hidden_dims]
```

Comment:
what's going on in this branch? `mask-1`?

Reply:
If there are context variables, the mask will have indices from 0 to (input_dim - 1). If there are no context variables, the mask will go from 1 to (input_dim - 1). The 0 index corresponds to the context variables, so this lets us build a representation of the context variables at each layer of the MADE in the conditional case

Reply:
nit: I think you can add a comment with the above explanation or link to Section 3.3 of https://arxiv.org/abs/1602.06701